packages/gatsby-transformer-mdx-introspection/src/jsx-ast-utils.js

```diff
@@ -0,0 +1,253 @@
+// Contains utility methods that deal directly with the JSX AST that is
+// outputted by Babel's parser
+
+/**
+ * Determines whether the object passed in has a type (which lets it )
+ * @param {object} value The object to examine
+ */
+function isAstNode(value) {
+  if (value == null || typeof value !== 'object') return false;
+  return Object.prototype.hasOwnProperty.call(value, 'type');
+}
+
+/**
+ * Gets all children nodes (nodes linked to by the current one).
+ * Note: does not check for specific properties, just aggregates all
+ * collections of linked nodes. May result in improper result if node
+ * contains any upreferences
+ * @param {object} node Babel JSX AST node
+ */
+function getChildren(node) {
+  let children = [];
+  if (node != null) {
+    // Examine each attribute of the object
+    Object.values(node).forEach(value => {
+      if (value != null && typeof value === 'object') {
+        if (Array.isArray(value)) {
+          // Check if it is a node array
+          if (value.every(isAstNode)) {
+            children = children.concat(value);
```

Comment:
What's the reason for the reassignment?

Reply:
The reassignment is used to merge the two arrays (children and value, which is resolved to be an node array in this path) using concat. The reason it's a reassignment and not a `value.forEach(v => children.push(v))` or a `push(...value)` is just to save on the iteration/spread cost and directly merge, but if you think it shouldn't use reassignment I don't mind changing it.